forgerock-auth/src/main/java/org/forgerock/android/auth/DefaultTokenManager.java

```diff
@@ -164,6 +164,10 @@ public void refresh(@NonNull AccessToken accessToken, final FRListener<AccessTok
             Listener.onException(listener, new AuthenticationRequiredException("Refresh Token does not exists."));
             return;
         }
+        //If the access token is not expired yet, revoke it
+        if (!accessToken.isExpired()) {
```

Comment:
is this check required ?  what happens when you try to revoke the expired token .  i had a android device in the past which my system time is always wrong due to calendar issues. so will there be a situation like server token is not expired and the client is expired with wrong calendar . its a edge case .

Reply:
Server returns the token "Expire In" which is the duration for the token, but not the exact time when the token will expired, client-side keep track of when the token expires, however, if the user change device time after receiving the token, the edge case scenario may happen. 

I put the logic to revoke the access token within the defined threshold, and avoid making additional network calls with the expired token. For e.g, if the access token has 30 min lifetime, and user set 10mins threshold, the access token still valid for 10mins.